### PR TITLE
[BEAM-1116] Support for new Timer API in Flink runner (streaming)

### PR DESCRIPTION
**Pull request by JingsongLi**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by JingsongLi** [comment]
Run Flink RunableOnService

**Comment by aljoscha** [comment]
I stared a Flink Runner post-commit hook here: https://builds.apache.org/view/Beam/job/beam_PostCommit_Java_RunnableOnService_Flink/1674/

**Comment by JingsongLi** [comment]
Thank you, it passed the test.

**Comment by aljoscha** [comment]
Perfect! Let's wait for an answer by @kennknowles to your question on the Jira Issa regarding how to extract the window from a namespace. Then I would merge this. 👍

**Comment by kennknowles** [comment]
Sorry for the delay! I missed the real comment between the bot noise on the JIRA. I answer there, but the basic thing is that namespaces are generalized beyond what user timers need. You only need `WindowNamespace`.

File: runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java [hunk 1]
```diff
@@ -538,9 +565,28 @@ public void onProcessingTime(InternalTimer<Object, TimerData> timer) throws Exce
     fireTimer(timer);
   }
 
+  /**
+   *  StateNamespace don't have a getWindow method.
+   */
+  private BoundedWindow getWindowFromNamespace(StateNamespace namespace) {
```
Comment (kennknowles) [inline runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java]: Here or in `fireTimer`, if these are really only user timers, that you can just `checkArgument(namespaces instanceof WindowNamespace)`.

**Comment by JingsongLi** [comment]
OK, I got it. User timers all come from StateNamespaces.window, so the namespace must be WindowNamespace.

**Pull request closed without merging**